pkgs/development/python-modules/gradio/default.nix

```diff
@@ -0,0 +1,204 @@
+    hatch-fancy-pypi-readme
+  ];
+
+  propagatedBuildInputs = [
+    aiofiles
+    aiohttp
+    altair
+    fastapi
+    ffmpy
+    markdown-it-py
+    mdit-py-plugins
+    markupsafe
+    matplotlib
+    numpy
+    orjson
+    pandas
+    pillow
+    pycryptodome
+    python-multipart
+    pydub
+    pyyaml
+    requests
+    uvicorn
+    jinja2
+    fsspec
+    httpx
+    pydantic
+    websockets
+    typing-extensions
+  ] ++ markdown-it-py.optional-dependencies.linkify;
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-asyncio
+    mlflow
+    #comet-ml # FIXME: enable once comet-ml is packaged
+    huggingface-hub
+    transformers
+    wandb
+    respx
+    scikit-image
+    ipython
+    ffmpeg
+    vega_datasets
+    boto3
+   ] ++ lib.optional shapWorks shap;
+
+  # Add pytest hook skipping tests that access network, marking them as "Expected fail" (xfail)
+  # We additionally xfail FileNotFoundError, as gradio often fail to upload test files to pypi.
+  preCheck = let
+    conftestSkipNetworkErrors = writeText "conftest-network-xfail.py" ''
+      from _pytest.runner import pytest_runtest_makereport as orig_pytest_runtest_makereport
+      import urllib, urllib3, httpx, requests, websockets
+
+      class NixNetworkAccessDeniedError(BaseException): pass
+      def deny_network_access(*a, **kw):
+        raise NixNetworkAccessDeniedError
+
+      def iterate_exc_chain(exc: Exception):
+        yield exc
+        if getattr(exc, "__context__", None) is not None:
+          yield from iterate_exc_chain(exc.__context__)
+        if getattr(exc, "__cause__", None) is not None:
+          yield from iterate_exc_chain(exc.__cause__)
+
+      httpx.Request = \
+      httpx.AsyncClient.get = \
+      httpx.AsyncClient.head = \
+      requests.head = \
+      requests.get = \
+      requests.post = \
+      urllib.request.urlopen = \
+      urllib.request.Request = \
+      urllib3.connection.HTTPSConnection._new_conn = \
+      websockets.connect = \
+        deny_network_access
+
+      def pytest_runtest_makereport(item, call):
+        tr = orig_pytest_runtest_makereport(item, call)
+        if call.excinfo is not None:
+          for exc in iterate_exc_chain(call.excinfo.value):
+            if isinstance(exc, NixNetworkAccessDeniedError):
+              tr.outcome, tr.wasxfail = 'skipped', "reason: Requires network access."
+            if isinstance(exc, FileNotFoundError):
+              tr.outcome, tr.wasxfail = 'skipped', "reason: Pypi dist bad."
+        return tr
+    '';
+  in ''
+    export HOME=$TMPDIR
+    cat ${conftestSkipNetworkErrors} >> test/conftest.py
+  '';
+
+  disabledTests = [
+    # Actually broken
+    "test_mount_gradio_app"
+
+    # FIXME: enable once comet-ml is packaged
+    "test_inline_display"
+    "test_integration_comet"
+
+    # Flaky (very sensitive to dep version)
+    "test_in_interface_as_output"
+    "test_should_warn_url_not_having_version"
+  ]
+  ++ lib.optionals (!shapWorks) [
```

Comment:
```suggestion
# shap breaking often makes this test flakey
```

Reply:
@pbsds isn't the boolean the opposite here ?
Maybe I'm understanding wrong here, but if shap works include the test, if it doesn't , then don't include it.
I added a suggestion so as to disable the test always to get rid completely of the shapworks evaluation.

Reply:
Yes, sorry